app/routes/reductions.js

```diff
@@ -0,0 +1,76 @@
+const organisationUnitConstants = require('../constants/organisation-unit')
+const reductionsService = require('../services/reductions-service')
+const Reduction = require('../services/domain/new-reduction')
+
+module.exports = function (router) {
+  router.get('/:organisationLevel/:id/reductions', function (req, res, next) {
+    var organisationLevel = req.params.organisationLevel
+    var id = req.params.id
+    var successText = req.body.successText ? req.params.successText : null
+
+    if (organisationLevel !== organisationUnitConstants.OFFENDER_MANAGER.name) {
+      throw new Error('Only available for offender manager')
+    }
+
+    var reductionsPromise = reductionsService.getReductions(id)
+
+    return reductionsPromise.then(function (result) {
+      return res.render('reductions', {
+        breadcrumbs: result.breadcrumbs,
+        linkId: id,
+        title: result.title,
+        active: [],
+        scheduled: [],
+        archived: [],
+        successText: successText
+      })
+    })
+  })
+
+  router.get('/:organisationLevel/:id/add-reduction', function (req, res, next) {
+    var organisationLevel = req.params.organisationLevel
+    var id = req.params.id
+
+    if (organisationLevel !== organisationUnitConstants.OFFENDER_MANAGER.name) {
+      throw new Error('Only available for offender manager')
+    }
+    var failureText = req.body.failureText ? req.params.failureText : null
+    var getAddReductionsReferenceDataPromise = reductionsService.getAddReductionsRefData(id, organisationLevel)
+
+    return getAddReductionsReferenceDataPromise.then(function (result) {
+      return res.render('add-reduction', {
+        breadcrumbs: result.breadcrumbs,
+        linkId: id,
+        title: result.title,
+        subTitle: result.subTitle,
+        referenceData: result.referenceData,
+        failureText: failureText
+      })
+    })
+  })
+
+  router.post('/:organisationLevel/:id/add-reduction', function (req, res, next) {
+    var organisationLevel = req.params.organisationLevel
+    var id = req.params.id
+
+    if (organisationLevel !== organisationUnitConstants.OFFENDER_MANAGER.name) {
+      throw new Error('Only available for offender manager')
+    }
+    var newReduction = new Reduction(
+      req.body.reasonForReduction, req.body.hours, req.body.reductionStartDate, req.body.reductionEndDate, req.body.notes
+    )
+    var addReductionsPromise = reductionsService.addReduction(id, newReduction)
+
+    return addReductionsPromise.then(function () {
```

Comment:
Do you need to check the returned id result here? Or will an error def be thrown on failure to insert reduction?

Reply:
An error will be thrown if there is an issue with inserting the data. We will do a validation check in the frontend and at the service layer. This way, we can be certain that the data going to the database is correct